pgjdbc/src/main/java/org/postgresql/util/LruCache.java

```diff
@@ -55,15 +57,15 @@ protected boolean removeEldestEntry(Map.Entry<Key, Value> eldest) {
         return false;
       }
 
-      Iterator<Map.Entry<Key, Value>> it = entrySet().iterator();
+      Iterator<Value> it = values().iterator();
       while (it.hasNext()) {
         if (size() <= maxSizeEntries && currentSize <= maxSizeBytes) {
           return false;
         }
 
-        Map.Entry<Key, Value> entry = it.next();
-        evictValue(entry.getValue());
-        long valueSize = entry.getValue().getSize();
+        Value value = it.next();
+        evictValue(value);
+        long valueSize = value.getSize();
```

Comment:
This seems like an unrelated change to "fix: LruCache evict on usage rather than insertion", meaning it should be in a separate commit. However, unlike a few of the others - this change is self-contained, and reduces complexity. Reduced complexity is good.